examples/function_calling/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/instructor-ai/instructor-go/pkg/instructor"
+	openai "github.com/sashabaranov/go-openai"
+)
+
+type SearchType string
+
+const (
+	Web   SearchType = "web"
+	Image SearchType = "image"
+	Video SearchType = "video"
+)
+
+type Search struct {
+	Topic string     `json:"topic" jsonschema:"title=Topic,description=Topic of the search,example=golang"`
+	Query string     `json:"query" jsonschema:"title=Query,description=Query to search for relevant content,example=what is golang"`
+	Type  SearchType `json:"type"  jsonschema:"title=Type,description=Type of search,default=web,enum=web,enum=image,enum=video"`
+}
+
+func (s *Search) execute() {
+	fmt.Printf("Searching for `%s` with query `%s` using `%s`\n", s.Topic, s.Query, s.Type)
+}
+
+type Searches = []Search
+
+func segment(ctx context.Context, data string) *Searches {
+
+	client, err := instructor.FromOpenAI[Searches](
```

Comment:
interesting so the [Searches] is on the client itself? Rather than 

client.CreateChatCompletion[Searches]

Reply:
Yeah maybe there is a better way to do it, but from what I tried you can't define an interface to use generic types (`Search` in this example) for the methods it has to implement. But yeah I would prefer it to be on the method and not the client so if I figure out how to do that better I will